### PR TITLE
fix(sitemap): rm trailing slash

### LINKED ISSUE #354: Sitemap slashes duplicated
Sitemap location url contains duplicate slashes, regardles of having trailing or no siteRoot slash.

Version in use 5.1.6.
`siteRoot: 'https://site.com'`

Expected:
`<loc>https://site.com/route/</loc>`

Result:
`<loc>https://site.com//route/</loc>`

### PR DESCRIPTION
**Pull request by davidpfahler**
Fixes #354.

## Changes/Tasks

- [x] Changed code

## Types of changes

- [ ] Refactoring/add tests (refactoring or adding test which isn't a fix or add a feature)
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly.
- [ ] My changes have tests around them

### CODE REVIEW

File: src/static/index.js [hunk 1]
```diff
@@ -452,7 +452,8 @@ export async function buildXMLandRSS ({ config }) {
   if (!siteRoot) {
     return
   }
-  const prefixPath = config.disableRoutePrefixing ? siteRoot : config.publicPath
+  let prefixPath = config.disableRoutePrefixing ? siteRoot : config.publicPath
+  prefixPath = prefixPath.replace(/\/$/, '')
   const xml = generateXML({
     routes: config.routes.filter(d => !d.is404).map(route => ({
       permalink: `${prefixPath}/${pathJoin(route.path)}`,
```
Comment (lottamus) [inline src/static/index.js]: You could still get a double slash here if `route.path === '/'`. Here is the code for `pathJoin`: https://github.com/nozzle/react-static/blob/master/src/utils/shared.js#L6

Perhaps `trimSlashes` should be broken into two functions: `trimLeadingSlashes` and `trimTrailingSlashes`. Then we can use those everywhere?

https://github.com/nozzle/react-static/blob/master/src/utils/shared.js#L52

Comment (tannerlinsley) [reply src/static/index.js]: Yeah, there are so many edge cases with slashes everywhere. I would much rather have all of them going through utility functions that have tests.

Comment (anjawaldbauer) [reply src/static/index.js]: Makes sense. Will update.

File: src/static/index.js [hunk 2]
```diff
@@ -455,7 +455,7 @@ export async function buildXMLandRSS ({ config }) {
   const prefixPath = config.disableRoutePrefixing ? siteRoot : config.publicPath
   const xml = generateXML({
     routes: config.routes.filter(d => !d.is404).map(route => ({
-      permalink: `${prefixPath}/${pathJoin(route.path)}`,
+      permalink: `${prefixPath}/${trimLeadingSlashes(pathJoin(route.path))}`,
```
Comment (lottamus) [inline src/static/index.js]: Still runs into the same issue if prefixPath ends in a slash. 

Needs 

```
permalink: `${trimTrailingSlashes(prefixPath)}/${trimLeadingSlashes(pathJoin(route.path))}`,
```

**Comment by davidpfahler** [comment]
@lottamus Thanks. Let me know if this is better

**Comment by davidpfahler** [comment]
Closing as this seems to be fixed in https://github.com/nozzle/react-static/commit/597f18f46a0d20d6663bc782b97f4641be7a35ed
